### ISSUE TITLE
Intranet only swagger editor via docker

### ISSUE THREAD

**Issue by Okeanos** [swagger-api/swagger-editor#1488]
| Q                                   | A
| ----------------------------------- | -------
| Bug or feature request?             |  feature
| Which Swagger/OpenAPI version?      | latest
| Which Swagger-Editor version?       | latest
| How did you install Swagger-Editor? |  docker
| Which broswer & version?            | any
| Which operating system?             |  any

### Expected Behavior
Using the docker image I should be able to start up a swagger editor instance that can be served from local non internet based resources only.

### Current Behavior
Using the (latest) docker image to start up a swagger editor instance I am greeted by an editor that requests a large amount of external internet based resources without (see swagger-editor/Dockerfile or swagger-editor/docker-run.sh) any way of telling the swagger editor to serve these resources from either the docker image itself or an additional resource like the swagger-codegen docker image. The Swagger UI can use the swagger-codegen docker image and some additional config values, though.

### Possible Solution
Add feature parity between swagger ui and swagger editor docker images concerning environment variables.

### Context
At company I am working we are currently trying to centrally host internal API documentations and would very much like to keep all related services such as the Swagger UI and Editor within the intranet only. This includes browser requests to fetch additional content such as stylesheets, JavaScript files, fonts, etc. when accessing these resources. The reasons for that should be fairly obvious. Right now there appears to be no point in using (the provided) docker images as they simply (out of the box and in case of the editor) rely on internet/third-party resources anyway. Maybe I am simply being stupid and am overlooking something very, very simple but after searching through all related issues, documentations, and related search results I could get my hands on I am stumped and can't find a way around this issue. Help would be very much appreciated.

**Comment by shockey** [comment]
Taking a look at what goes into the Docker image (`index.html` in particular), I didn't see any references to any assets that would require an internet connection (see: https://github.com/swagger-api/swagger-editor/blob/master/index.html)

Can you provide the errors you're getting when running the image? Seeing what it's trying to reach out for would be very helpful.

**Comment by Okeanos** [comment]
Okay, maybe I phrased this slightly wrong initially / from a wrong perspective. The Swagger Editor fetches at least three different resources from generator.swagger.io on load even when supplied with a local API file. These are `servers`. `clients`, and `swagger.json`. These may or may not be necessary for the editor to work as such. 

My problem stems from the fact that any user which uses such an intranet hosted editor instance who (for whatever reason, e.g. a local proxy) also has internet access may unknowingly or inadvertently submit API file names and/or contents to swagger.io by using the generator features. Using the standard Docker image I don't see a way to easily disable or reconfigure these extended features. Whether or not the people running and maintaining swagger.io actually do anything with that information is irrelevant I think.

Given that additional Docker images for the validator and the codegen/generator features exist I thought that the Editor could be provisioned similarly to the UI.

**Comment by shockey** [comment]
Ah, got it now. Those three requests are needed for the codegen features (`Generate Server` and `Generate Client` in the top menu) to work. No API definition information is sent in those initial requests, but if the Codegen features are used, the definition is sent to `generator.swagger.io`.

While it is possible to disable the Codegen feature if you're controlling the initial Swagger-Editor call, there's currently no way to disable it from the Docker level. 

> Given that additional Docker images for the validator and the codegen/generator features exist I thought that the Editor could be provisioned similarly to the UI.

The Validator project is actually different from the validation you find in the Editor :smile: Similarly, the Codegen feature in the Editor that you're observing is an interface to a running instance of the Codegen project.

---

I think the best way forward here would be to add a configuration option, `codegenUrl`, that is controllable through Docker environment variables.

Pinging @webron for prioritization and thoughts!

**Comment by webron** [comment]
Sure, a configuration option makes sense and it existed in the previous version of the editor as well. As for prioritization, given what we have on our plate right now, not very high.

**Comment by shockey** [comment]
cc: #1547

**Comment by Kallb123** [comment]
Being able to define a local codegen server seems incredibly useful. In order to use Swagger, we will have to use the editor, export as JSON and then use Postman to send the spec to the codegen server.

**Comment by hvindin** [comment]
Due to concerns that our one of the internal security teams had about people inside the org I'm currently with using a tool hosted off-site to edit internal swagger definitions we've actually ended up just hacking the swagger editor/validator/generator etc. by basically just downloading the latest versions of the docker files, using our highly sophisticated `sed` and `grep` skills to replace anything outside the internal network with an endpoint that will be hosting the equivalent thing internally, and then just running all of that.

It works OK but is pretty hacky and painful, I know I'd very much appreciate the ability to build the distributable files with overridden URL's.

And it _feels_ like it wouldn't be that difficult a feature to add, I've just managed to get lost in the code base a few times and haven't actually invested the time to go and figure out where to make the changes.
